### PR TITLE
Create `HiddenTextExpander` component

### PR DESCRIPTION
**Pull request by manuelpuyol**
Part of https://github.com/primer/view_components/issues/443.

Creates `HiddenTextExpander` component based on https://primer.style/css/components/buttons#hidden-text-button

image

### CODE REVIEW

File: app/components/primer/hidden_text_expander.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    #   <%= render(Primer::HiddenTextExpander.new) %>
+    #
+    # @example Inline
+    #   <%= render(Primer::HiddenTextExpander.new(inline: true)) %>
+    #
+    # @example Styling the button
+    #   <%= render(Primer::HiddenTextExpander.new(button_arguments: { p: 1, classes: "my-custom-class" })) %>
+    #
+    # @param inline [Boolean] Whether or not the expander is inline.
+    # @param button_arguments [Hash] <%= link_to_system_arguments_docs %> for the button element.
```
[the remark this answers is not in the document]
Comment (manuelpuyol) [reply app/components/primer/hidden_text_expander.rb:16]: yes! we set some `js-` classes, `test_selector` and `color`

File: app/components/primer/hidden_text_expander.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  # Use HiddenTextExpander to indicate and toggle hidden text.
+  class HiddenTextExpander < Primer::Component
+    # @example Default
+    #   <%= render(Primer::HiddenTextExpander.new) %>
+    #
+    # @example Inline
+    #   <%= render(Primer::HiddenTextExpander.new(inline: true)) %>
+    #
+    # @example Styling the button
+    #   <%= render(Primer::HiddenTextExpander.new(button_arguments: { p: 1, classes: "my-custom-class" })) %>
```
Comment (joelhawksley) [inline app/components/primer/hidden_text_expander.rb:13]: Would it be possible for the examples to actually toggle hidden text?

Comment (manuelpuyol) [reply app/components/primer/hidden_text_expander.rb:13]: not now, it requires some JS behaviors that we can explore adding in the future

Comment (joelhawksley) [reply app/components/primer/hidden_text_expander.rb:13]: 🤔 gotcha. I think we should add JS behaviors to the Beta criteria then 😄

Comment (manuelpuyol) [reply app/components/primer/hidden_text_expander.rb:13]: yeah, but I don't think this exact component should have the behavior, it is part of a bigger component problably

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**